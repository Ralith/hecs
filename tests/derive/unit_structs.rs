use hecs::Bundle;

#[derive(Bundle)]
struct NamedUnit {}

#[derive(Bundle)]
struct TupleUnit();

#[derive(Bundle)]
struct Unit;

fn main() {}
