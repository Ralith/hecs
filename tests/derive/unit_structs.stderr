error[E0282]: type annotations needed for `&T`
 --> $DIR/unit_structs.rs:3:10
  |
3 | #[derive(Bundle)]
  |          ^^^^^^ consider giving this closure parameter the explicit type `&T`, where the type parameter `T` is specified
  |
  = note: type must be known at this point
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0609]: no field `0` on type `&_`
 --> $DIR/unit_structs.rs:3:10
  |
3 | #[derive(Bundle)]
  |          ^^^^^^
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0282]: type annotations needed for `&T`
 --> $DIR/unit_structs.rs:6:10
  |
6 | #[derive(Bundle)]
  |          ^^^^^^ consider giving this closure parameter the explicit type `&T`, where the type parameter `T` is specified
  |
  = note: type must be known at this point
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0609]: no field `0` on type `&_`
 --> $DIR/unit_structs.rs:6:10
  |
6 | #[derive(Bundle)]
  |          ^^^^^^
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0282]: type annotations needed for `&T`
 --> $DIR/unit_structs.rs:9:10
  |
9 | #[derive(Bundle)]
  |          ^^^^^^ consider giving this closure parameter the explicit type `&T`, where the type parameter `T` is specified
  |
  = note: type must be known at this point
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0609]: no field `0` on type `&_`
 --> $DIR/unit_structs.rs:9:10
  |
9 | #[derive(Bundle)]
  |          ^^^^^^
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
